skribenten-web/frontend/src/Brevredigering/LetterEditor/actions/toggleBulletList.ts

```diff
@@ -0,0 +1,436 @@
+
+    if (current.type === "ITEM_LIST") {
+      if (hasNext) {
+        temp.push(current);
+      } else {
+        const ids = temp.filter((il) => !!il.id).map((il) => il.id!);
+        result.newcontent.push(
+          newItemList({
+            id: ids.length > 0 ? ids[0] : null,
+            items: [...temp.flatMap((i) => i.items), ...current.items],
+            deletedItems: temp
+              .slice(1)
+              .filter((i) => !!i.id)
+              .map((i) => i.id!),
+          }),
+        );
+      }
+    } else {
+      if (temp.length > 0) {
+        const ids = temp.filter((il) => !!il.id).map((il) => il.id!);
+        result.newcontent.push(
+          newItemList({
+            id: ids.length > 0 ? ids[0] : null,
+            items: temp.flatMap((i) => i.items),
+            deletedItems: temp
+              .slice(1)
+              .filter((i) => !!i.id)
+              .map((i) => i.id!),
+          }),
+        );
+        result.newDeleted.push(...ids.slice(1).filter((id) => id !== null));
+        temp = [];
+        result.newcontent.push(current);
+      } else {
+        result.newcontent.push(current);
+      }
+    }
+  }
+
+  return result;
+};
+
+/**
+ * Når vi fjerner et punkt fra en punktliste, så må man ta høyde for at man kan potensielt ha content før og etter punktlisten.
+ * Vi må også ta høyde for at vi kan være på første, siste eller et sted i mellom punktene i listen.
+ *  Det er fordi vi må styre hvor den nye blocken skal havne, samt resten av innholdet i punktlisten.
+ *
+ * Fordi vi gjør en såpass stor endring i dokument strukturen, Så må vi oppdatere fokuset til editorstaten til å være på rett plass
+ */
+const toggleBulletListOff = (draft: Draft<LetterEditorState>, itemContentIndex: ItemContentIndex) => {
+  const thisBlock = draft.redigertBrev.blocks[itemContentIndex.blockIndex];
+  const thisItemList = thisBlock.content[itemContentIndex.contentIndex] as ItemList;
+
+  if (itemContentIndex.itemIndex === 0) {
+    toggleBulletListOffAtTheStartOfItemList({ draft, itemContentIndex });
+  } else if (itemContentIndex.itemIndex > 0 && itemContentIndex.itemIndex < thisItemList.items.length - 1) {
+    toggleBulletListOffBetweenListElements({ draft, itemContentIndex });
+  } else if (itemContentIndex.itemIndex === thisItemList.items.length - 1) {
+    toggleBulletListOffAtTheEndOfItemList({ draft, itemContentIndex });
+  }
+};
+
+/**
+ * Fjerner det første punktet fra en punktliste
+ * Bygger dette punktet som en ny block, og setter den inn i dokumentet
+ * Legger tilbake resten av punktene i punktlisten
+ *
+ * Fordi vi gjør en såpass stor endring i dokument strukturen, Så må vi oppdatere fokuset til editorstaten til å være på rett plass
+ */
+const toggleBulletListOffAtTheStartOfItemList = (args: {
+  draft: Draft<LetterEditorState>;
+  itemContentIndex: ItemContentIndex;
+}) => {
+  const prevBlocks = args.draft.redigertBrev.blocks.slice(0, Math.max(0, args.itemContentIndex.blockIndex));
```

Comment:
Har denne blitt litt mer komplisert enn nødvendig? Jeg tror ikke det er nødvendig å tre ut utenfor gjeldende blokk (`args.itemContentIndex.blockIndex`). Burde bare være å flytte innholdet fra item ut i blokken før itemList, og tilsvarende for om toggling skjer på siste item. Om det er i midten så må itemList splittes i to i gjeldende blokk, med innholdet fra togglet item i mellom de to itemlistene.